.github/workflows/main.yaml

```
@@ -4,38 +4,49 @@
 
     permissions:
       contents: read
-      packages: write  # Added permission to push Docker images
+      packages: write  
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@cbb722410c2e876e24abbe8de2cc27693e501dcb # v2
-      - name: Set up Docker
-        uses: docker/setup-buildx-action@c47758b77c9736f4b2ef4073d4d51994fabfe349 # v3.7.1
+        uses: actions/checkout@v2
```

actionlint

the runner of "actions/checkout@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)